searx/poolrequests.py

```diff
@@ -233,3 +426,96 @@ def patch(url, data=None, **kwargs):
 
 def delete(url, **kwargs):
     return request('delete', url, **kwargs)
+
+
+def init():
+    # log
+    for logger_name in ('hpack.hpack', 'hpack.table'):
+        logging.getLogger(logger_name).setLevel(logging.WARNING)
+
+    # loop
+    def loop_thread():
+        global LOOP
+        LOOP = asyncio.new_event_loop()
+        LOOP.run_forever()
+
+    th = threading.Thread(
+        target=loop_thread,
+        name='asyncio_loop',
+        daemon=True,
+    )
+    th.start()
+
+
+@atexit.register
+def done():
+    """Close all HTTP client
+
+    Avoid a warning at exit
+    """
+    global LOOP
+
+    async def close_client(client):
+        try:
+            await client.aclose()
+        except httpx.HTTPError:
+            pass
+
+    async def close_clients():
+        await asyncio.gather(*[close_client(client) for client in CLIENTS.values()], return_exceptions=False)
+    future = asyncio.run_coroutine_threadsafe(close_clients(), LOOP)
+    # wait 3 seconds to close the HTTP clients
+    future.result(3)
+
+
+init()
+
+
+# ## TEMPORARY DEBUG ##
```

Comment:
The following lines will be removed later.
Perhaps kept in a debug plugin.

Reply:
Oh, I got it. The endpoint will be removed as well I assume.

Reply:
Removed